tests/Functional/FileFromMemoryTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Mailgun\Tests\Functional;
+
+/**
+ * Add attachment with file from memory
+ *
+ * @author Wim Verstuyf <wim.verstuyf@codelicious.be>
+ */
+class FileFromMemoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testAddFileFromMemory()
+    {
+        $fileValidator = function($files) {
+            $attachments = array_filter($files, function($f) {
+                return strpos($f['name'], 'attachment') !== false;
+            });
+
+            $attachments = array_map(function($f) {
+                return [
+                    'name' => $f['name'],
+                    'contents' => is_resource($f['contents'])?fread($f['contents'], 50):$f['contents'],
```

Comment:
That correct... Hm that is weird.. I do not think we should allow the 'contents' to be both a resource and string. 

We should expect only resources here to at least have a decent API.